### PR TITLE
Improve the upgrade configuration system

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Updating the upgrade configuration system
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | The different configurations work the same way as before. It is now possible to add configurations to a CLI upgrade

### CODE REVIEW

File: classes/Parameters/UpgradeConfiguration.php [hunk 1]
```diff
@@ -160,6 +184,20 @@ public function shouldUpdateRTLFiles(): bool
...
     }
 
+    public static function shouldDisabledOverride(): bool
+    {
+        return (bool) Configuration::get('PS_DISABLE_OVERRIDES');
+    }
+
+    public static function updateDisabledOverride(bool $value, ?int $shopId = null): void
+    {
+        if ($shopId) {
```
Comment (kpodemski) [inline classes/Parameters/UpgradeConfiguration.php:194]: It probably doesn't make sense to rely on `$shopId` in this module, you work on one PrestaShop instance, not on a single store.

Comment (M0rgan01) [reply classes/Parameters/UpgradeConfiguration.php:194]: @kpodemski 

I took this configuration which was already present at the base. I tested to make sure I understand why this configuration was added. 

It is possible to set up this configuration in a particular shop in the advanced settings. 

When adding this configuration in the autoupgrade module, this feature with the context of a shop was also added, to keep the same value on this configuration between the 2 pages.

File: classes/Parameters/UpgradeConfiguration.php [hunk 2]
```diff
@@ -160,6 +190,28 @@ public function shouldUpdateRTLFiles(): bool
         return (bool) $this->get('PS_AUTOUP_UPDATE_RTL_FILES');
     }
 
+    public static function shouldDisabledOverride(): bool
```
Comment (Quetzacoalt91) [inline classes/Parameters/UpgradeConfiguration.php:193]: Let's find another name for this method, to replace the "should" and fix the verb in the past tense.

* `isOverrideAllowed()`
* `canOverrideBeUsed()`
* `isOverridingEnabled()`

File: classes/Parameters/UpgradeConfiguration.php (hunk 1 above)
Comment (Quetzacoalt91) [reply classes/Parameters/UpgradeConfiguration.php:200]: Disabling overrides makes sense to try the shop after upgrade. Unfortunately the global value can be overriden if a specific shop has its own configuration value.

The current implementation covers this case.

File: classes/Task/Miscellaneous/UpdateConfig.php [hunk 3]
```diff
@@ -131,11 +132,16 @@ public function run(): int
...
             $config['directory.version_num'] = $configurationData['directory_num'];
         }
-        if (isset($configurationData['skip_backup'])) {
-            $config['skip_backup'] = $configurationData['skip_backup'];
-        }
-        if (isset($configurationData['PS_AUTOUP_CHANGE_DEFAULT_THEME'])) {
-            $config['PS_AUTOUP_CHANGE_DEFAULT_THEME'] = $configurationData['PS_AUTOUP_CHANGE_DEFAULT_THEME'];
+
+        foreach (UpgradeConfiguration::UPGRADE_CONST_KEYS as $key) {
+            if (isset($configurationData[$key])) {
```
Comment (Quetzacoalt91) [inline classes/Task/Miscellaneous/UpdateConfig.php:137]:
```suggestion
            if (!isset($configurationData[$key])) {
                continue;
            }
```

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by AureRita — DISMISSED** [review]

**Pull request merged**